packages/java/src/node/java-service.ts

```diff
@@ -0,0 +1,67 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { NodePackage, ExtensionPackage } from '@theia/application-package';
+import { JavaService } from '@theia/java/src/common/java-common';
+import { ApplicationPackageManager } from '@theia/application-manager';
+import { injectable } from 'inversify';
+import * as paths from 'path';
+
+interface JavaContributes extends NodePackage {
+    javaExtensions?: string[];
```

Comment:
This needs documentation. How does this work? I assume I need to add some text to package.json? What is the syntax of the strings?